great_expectations/datasource/fluent/spark_datasource.py

```diff
@@ -134,7 +134,7 @@ def _validate_batch_request(self, batch_request: BatchRequest) -> None:
                 datasource_name=self.datasource.name,
                 data_asset_name=self.name,
                 options={},
-                batch_slice=batch_request.batch_slice,
+                batch_slice=batch_request.batch_slice_input,  # type: ignore[attr-defined]
```

Comment:
we should not have "batch_slice_input" as n attribute, and certainly not a public one. What is it required?

Reply:
This is how we are reversing the serialization of the input strings, ints, lists, etc. I want to make this (more) private, but pydantic does weird things when you add the proceeding underscore to variables. We are only using this internally and it requires the type ignore you see on this line. This attribute is not included in the stub file, so it will not show up in autocomplete.

Reply:
I was able to figure this out!